fearless/Modules/Staking/Services/EraValidatorsService/EraValidatorService+Fetch.swift

```diff
@@ -310,23 +296,23 @@ extension EraValidatorService {
     }
 
     private func updateIfNeeded(
-        chain: Chain,
         activeEra: UInt32,
         prefixKey: Data,
         codingFactory: RuntimeCoderFactoryProtocol
     ) {
-        guard activeEra == self.activeEra, chain == self.chain else {
+        guard activeEra == self.activeEra else {
             logger?.warning("Update triggered but parameters changed. Cancelled.")
             return
         }
 
-        guard let localFactory = try? ChainStorageIdFactory(chain: chain) else {
-            logger?.error("Can't create local factory")
+        guard let localPrefixKey = try? createLocalExposurePrefixKey(
+            for: chainId,
+            activeEra: activeEra
+        ) else {
+            logger?.error("Can't create local storage key prefix key")
```

Comment:
Because of errors I don't like how we declare logger to be Optional and give it `nil` default value. It is easy to forget to pass an actual logger instance and no logs will be produced. Can we discuss alternatives?